perfkitbenchmarker/linux_benchmarks/object_storage_service_benchmark.py

```diff
@@ -1037,13 +1036,16 @@ def MultiStreamReadBenchmark(results, metadata, vms, command_builder,
         "object_storage_read_objects file specified requires exactly %d " \
         "VMs, but %d were provisioned." % (len(read_objects), len(vms))
     for vm, vm_objects_written in zip(vms, read_objects):
-      # Note that this overwrites the same local file over and over, because
-      # each VM's worker process expects it to have the same file name.
-      tmp_objects_written_path = os.path.join(temp_dir.GetRunDirPath(),
-                                              OBJECTS_WRITTEN_FILE)
+      # Note that each file is written with a unique name so that parallel runs
+      # don't overwrite the same local file. They are pushed to the VM to a file
+      # named OBJECTS_WRITTEN_FILE.
+      tmp_objects_written_path = os.path.join(vm_util.GetTempDir(),
+                                              '%s-%s' % (OBJECTS_WRITTEN_FILE,
+                                                         vm.name))
       with open(tmp_objects_written_path, 'w') as objects_written_file:
         objects_written_file.write(json.dumps(vm_objects_written))
-      vm.PushFile(tmp_objects_written_path, '/tmp/pkb/')
+      vm.PushFile(tmp_objects_written_path,
+                  os.path.join('/tmp/pkb/', OBJECTS_WRITTEN_FILE))
```

Comment:
Looks like it was this way before this change, but I think using the vm_util.VM_TMP_DIR constant is preferred to '/tmp/pkb/' when referring to the temp directory on remote machines. Also, since this is a benchmark for linux VMs, it should use posixpath.join instead of os.path.join for paths on remote hosts since os.path.join will use "\\" if PKB is running on a windows machine

Reply:
Done.